File: buildSrc/src/main/java/org/eclipse/dataspaceconnector/gradle/DataspaceConnectorPlugin.java
package org.eclipse.dataspaceconnector.gradle;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class DataspaceConnectorPlugin implements Plugin<Project> {

    @Override
    public void apply(Project project) {
        var extension = project.getExtensions().create(
            "dataspaceconnectorplugin", DataspaceConnectorPluginExtension.class
        );
        project.getSubprojects().forEach(p -> registerTask(p, extension));
    }

    private void registerTask(Project project, DataspaceConnectorPluginExtension extension) {
        project.getTasks()
            .register("applyDependencyRules", DependencyRulesTask.class)
            .configure(task -> task.getFailOnError().set(extension.isFailSeverity()));
    }
}

File: buildSrc/src/main/kotlin/DataspaceConnectorPlugin.kt
package org.eclipse.dataspaceconnector.gradle;

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.kotlin.dsl.register

open class DataspaceConnectorPlugin : Plugin<Project> {

    override fun apply(project: Project) {
        project.subprojects.forEach { p -> registerTask(p) }
    }

    private fun registerTask(project: Project) {
        project.tasks.register<DependencyRulesTask>("applyDependencyRules") {
            dependencyAnalysis.set("warn") // TODO: initialize from parameter
        }
    }
}